docs/conf.py

```diff
@@ -7,11 +7,12 @@
 release = ""
 language = None
 master_doc = "index"
-extensions = ["sphinx.ext.autodoc", "sphinx.ext.viewcode"]
+extensions = ["sphinx.ext.autodoc", "sphinx.ext.autodoc.typehints", "sphinx.ext.viewcode"]
 source_suffix = [".rst", ".md"]
 exclude_patterns = ["_build", "Thumbs.db", ".DS_Store"]
 pygments_style = "sphinx"
-autodoc_typehints = "description"
+autoclass_content = "both"
+autodoc_typehints = "both"
```

Comment:
Just calling out so it doesn't get burried in the PR description, this requires Sphinx 4.1 and probably shouldn't be changed on older versions.